### PR TITLE
[#2504] Adding architecture documentation

### PR DESCRIPTION
**Pull request by alextreme**

### CODE REVIEW

File: docs/architecture/index.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+Accounts app
+.. graphviz:: accounts.dot
+
+Configuration app	      
+.. graphviz:: configurations.dot
+
+Haal Centraal app
+.. graphviz:: haalcentraal.dot
+
+KvK app
```
Comment (jiromaykin) [inline docs/architecture/index.rst:65]: Perhaps not vital but "KVK" is now officially spelled with all-caps, instead of the old lower case little 'v'.

**Review by jiromaykin — CHANGES_REQUESTED** [review]
I could approve but just wondering: Should these docs use the Dimpact logo?
open-Formulieren has its own logo in their docs:
 ⬇️
 
Screenshot 2024-06-10 at 12 36 09

File: docs/architecture/index.rst (hunk 1 above)
Comment (alextreme) [reply docs/architecture/index.rst]: Thanks!

**Comment by alextreme** [comment]
> I could approve but just wondering: Should these docs use the Dimpact logo?
> open-Formulieren has its own logo in their docs:

Added https://taiga.maykinmedia.nl/project/open-inwoner/task/2543 for this, we don't have a logo atm and the one currently used is originally from i4Sociaal

File: docs/architecture/index.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+The Django code-base is structured into a range of built-in Django apps, one for each integration (`haalcentraal`, `openzaak`, `openklant` etc).
+
+For making use of these integrations in a pluggable fashion, OIP uses Django CMS apphooks which can be found in the `cms` app:
+
+ * `products` publishes the PDC / Category and Product details to anonymous and logged-in users (eg. Productenpagina's)
+ * `cases` exposes the Open Zaak case details to logged-in users (eg. Mijn aanvragen)
+ * `collaborate` exposes the cooperative planning module to logged-in users (eg. Samenwerken)
+ * `inbox` exposes the internal messaging app to logged-in users (eg. Mijn Berichten)
+ * `profile` exposes the profile to logged-in users (eg. Mijn Profiel)
+ * `benefits` exposes the ability to download benefit specifications via SSD to logged-in users(eg. Mijn Uitkeringen)
```
[the remark this answers is not in the document]
Comment (alextreme) [reply docs/architecture/index.rst:38]: Took a while for me to grok this one, but for the ones 'in the know' (eg. municipalities) it's clear

**Comment by alextreme** [comment]
Thanks Sidney for the tweaks: merged in the suggestions, added the updated logo and rebased

File: docs/architecture/index.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+With OIP we make use of a number of shared Maykin-built Django apps like django-digid-eherkenning, zgw-consumers, notifications-api-common, maykin-2fa, django-admin-index, django-open-forms-client, django-log-outgoing-requests, django-setup-configuration and mozilla-django-oidc-db.
+
+
+Datamodels
+==========
+
+Generated based on OIP v1.17
+
+Accounts app
+.. graphviz:: accounts.dot
```
Comment (pi-sigma) [inline docs/architecture/index.rst:51]: The links don't work when I build the docs locally. I need to add an empty line between heading and link. For readability, I suggest adding styling to the subheadings + one empty line between each component. 

```
Datamodels
==========

Generated based on OIP v1.17

Accounts app
""""""""""""

.. graphviz:: accounts.dot

|

Configuration app
"""""""""""""""""

.. graphviz:: configurations.dot

|

Haal Centraal app
"""""""""""""""""

.. graphviz:: haalcentraal.dot

|

KVK app
"""""""

.. graphviz:: kvk.dot

|

Open Klant app
""""""""""""""

.. graphviz:: openklant.dot

|

Open Zaak app
"""""""""""""

.. graphviz:: openzaak.dot

|

PDC app
"""""""

.. graphviz:: pdc.dot

|

Plans app
"""""""""

.. graphviz:: plans.dot

|
```

**Review by pi-sigma — CHANGES_REQUESTED** [review]
Links don't work (see comment). Let me know if you don't have this issue.

**Review by pi-sigma — APPROVED** [review]

**Review by jiromaykin — APPROVED** [review]

**Pull request merged**